### DISCUSSION TITLE
JK BMS

### DISCUSSION THREAD

**Discussion by Mike-Bou** [Louisvdw/dbus-serialbattery#323] [Q&A] [answered]
i get this output from current log
~~~
@40000000638fd49309763bcc INFO:SerialBattery:Testing Daly
@40000000638fd49319592c84 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4931966fbac INFO:SerialBattery:Testing Daly
@40000000638fd49329422c04 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd493295156d4 INFO:SerialBattery:Testing Jkbms
@40000000638fd493392dac24 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd493393b67c4 INFO:SerialBattery:Testing Lifepower
@40000000638fd4940d8aba6c ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4940d982404 ERROR:SerialBattery:>>> ERROR: Incorrect Data
@40000000638fd4940da05994 INFO:SerialBattery:Testing Renogy
@40000000638fd4941d7dcff4 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4941d8ba304 INFO:SerialBattery:Testing Renogy
@40000000638fd4942d67f46c ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4942d766b8c INFO:SerialBattery:Testing Ecs
@40000000638fd495131e9994 INFO:SerialBattery:Testing LltJbd
@40000000638fd49522fe9094 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd495230c8e9c INFO:SerialBattery:Testing Ant
@40000000638fd49532ec5aa4 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd49532f9b49c ERROR:SerialBattery:>>> ERROR: Incorrect Data
@40000000638fd4953301caec INFO:SerialBattery:Testing Daly
@40000000638fd496076ea42c ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd496076ec36c INFO:SerialBattery:Testing Daly
@40000000638fd4961745c574 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4961753a054 INFO:SerialBattery:Testing Jkbms
@40000000638fd496274e5314 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd496275c64a4 INFO:SerialBattery:Testing Lifepower
@40000000638fd496374e0afc ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd496375b64f4 ERROR:SerialBattery:>>> ERROR: Incorrect Data
@40000000638fd49637638ae4 INFO:SerialBattery:Testing Renogy
@40000000638fd4970ba3ce14 ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4970bb162a4 INFO:SerialBattery:Testing Renogy
@40000000638fd4971b8e293c ERROR:SerialBattery:>>> ERROR: No reply - returning
@40000000638fd4971bfd274c INFO:SerialBattery:Testing Ecs
@40000000638fd498019b79e4 ERROR:SerialBattery:ERROR >>> No battery connection at /dev/ttyUSB1
~~~

The RS485 converter im using is FTDI, and both comms lights flash.
Seems like there is some info, reported as Incorrect Data, for the wrong type of BMS.

I have JK 2Amp balance 4-8S 200Amp BMS, with 7S. Victron 100-50 SCC

I do not have a Victron Energy Inverter, so no VE Bus showing on Raspberry Pi. Maybe that is the problem?
Using Venus 3.x large.

**Comment by Louisvdw** [comment]
You don't need an inverter. The battery will still connect and work just with the GX device (your Pi in this case)
The issue most of the time is with the cabling and connectors.
If you are using a RS485 converter I assume that you also have the JKBMS RS485 adapter which is plugged into the GPS port on the BMS, and then you have you FTDI RS485 adaptor connected to that and into the USB port of the Pi. Is this how your setup is connected?

**Reply by Mike-Bou** [reply]
Yes,  that is the right connections. For the RaspberryPi end, i chose device DT5019, with standard USB plug, on 18" cable to a RS232/422/485 unit, which has the lights, and  a 9 pin D Type plug, to a wiring terminal breakout board. it has 5 terminals. just tried removing the ground terminal from the BMS, still not working. The RS485 is shown as 2 wire connection: 1 = T/R+ on the 'D board', to 485+ from the JK BMS RS485 adaptor. 2 =T/R- on the 'D board'  to 485- from th JK RS485 adaptor. i first wired them the other way, and only one light was flashing. The other outputs of the DT5019 are labelled RXD+ & RXD- used for RS422. Maybe i need a different device, but here in Philippines, options are very limited.

**Reply by Louisvdw** [reply]
I think the problem might be your adapter. I see you say it has a FTDI chip, but none of the source I could find can confirm that.
I could be possible to to make it work, but you need to set up a driver for that device in your VenusOS (not something I have information to assist with). I know that FTDI and CH340 based chips have drivers as part of VenusOS and they do work.

**Reply by Mike-Bou** [reply]
OK Louis, thank you very much for the replies. Looks like you have created an excellent product for connecting BMS to Venus. i will try to get another USB to RS485, such as the Waveshare, to get it working. Longer term, i am going to upgrade the BMS to Electrodacus SBMS0, with external shunts.

**Reply by fast-quality** [reply]
Hello Mike-Bou,

I´m new in the Thematik, but I have read a lot last time. My decission was to make the connection only by using a ttl to usb adapter connect directly to the bms.
Additionally, so that no Problems arise by grind mass, I have connected a usb galvanik divider between. 
Many People have Problems with the original JK RS485 because of grind mass, that can destroy the adapter.

**Comment by Mike-Bou** [comment, marked as answer]
I got it working, great, with a new RS485 to USB, the cheap transparent blue type. Good job you did, writing this, thanksSent from my iPhoneOn 10 Dec 2022, at 04:12, fast-quality ***@***.***> wrote:﻿
Hello Mike-Bou,
I´m new in the Thematik, but I have read a lot last time. My decission was to make the connection only by using a ttl to usb adapter connect directly to the bms.
Additionally, so that no Problems arise by grind mass, I have connected a usb galvanik divider between.
Many People have Problems with the original JK RS485 because of grind mass, that can destroy the adapter.

**Reply by Mike-Bou** [reply]
i just left off the ground wire connection from the JK RS485 adaptor to the USB adaptor
